sdk/include/reactos/shlobj_undoc.h

```diff
@@ -528,11 +528,11 @@ DECLARE_INTERFACE_(INamespaceProxy, IUnknown)
 	STDMETHOD_(ULONG,AddRef)(THIS) PURE;
 	STDMETHOD_(ULONG,Release)(THIS) PURE;
 	 /*** INamespaceProxy ***/
-	STDMETHOD(GetNavigateTarget)(THIS_ long paramC, long param10, long param14) PURE;
-	STDMETHOD(Invoke)(THIS_ long paramC) PURE;
-	STDMETHOD(OnSelectionChanged)(THIS_ long paramC) PURE;
-	STDMETHOD(RefreshFlags)(THIS_ long paramC, long param10, long param14) PURE;
-	STDMETHOD(CacheItem)(THIS_ long paramC) PURE;
+	STDMETHOD(GetNavigateTarget)(THIS_ IN PCIDLIST_ABSOLUTE pidl, OUT PIDLIST_ABSOLUTE ppidlTarget, OUT ULONG *pulAttrib) PURE;
+	STDMETHOD(Invoke)(THIS_ IN PCIDLIST_ABSOLUTE pidl) PURE;
+	STDMETHOD(OnSelectionChanged)(THIS_ IN PCIDLIST_ABSOLUTE pidl) PURE;
+	STDMETHOD(RefreshFlags)(THIS_ OUT DWORD *pdwStyle, OUT DWORD *pdwExStyle, OUT DWORD *dwEnum) PURE;
+	STDMETHOD(CacheItem)(THIS_ IN PCIDLIST_ABSOLUTE pidl) PURE;
```

Comment:
Could you use the new SAL2 annotations?

Reply:
There is no "optionality" info for SAL2.

Reply:
what do you mean by "optionality" info? Of course there is.

Reply:
I don't know whether "Is the parameter optional?" or not.

Reply:
What do you mean? You don't know whether you should use e.g. `_In_` or `_In_opt_` , for example?

Reply:
`_In_` means non-optional input.
`_In_opt_` means optional input.
Microsoft didn't provide optional info.

Reply:
> `_In_` means non-optional input. `_In_opt_` means optional input.

In the old-school annotations, this is the same. If you don't add `OPTIONAL` it would also mean non-optional input (or output).

Reply:
> Because they aren't optional.

Ok, I will fix them tomorrow.

Reply:
Fixed.